CrossSectionAnalysis/CrossSectionAnalysis.py

```diff
@@ -0,0 +1,422 @@
+    if self.logic.pathArray.size > 1:
+        sliderWidget.maximum = (self.logic.pathArray.size / 3) - 1 - 1
+    
+  # logic.onWidgetMarkupPointAdded gets called first
+  def onWidgetMarkupPointAdded(self, caller, event):
+    self.setSliderWidget()
+    self.ui.positionIndexSliderWidget.setValue(0)
+    self.showCurrentPositionData(0)
+
+  def onWidgetMarkupPointRemoved(self, caller, event):
+    self.setSliderWidget()
+    self.ui.positionIndexSliderWidget.setValue(0)
+    self.showCurrentPositionData(0)
+  
+  # Position on path is not reset here
+  def onWidgetMarkupPointEndInteraction(self, caller, event):
+    self.showCurrentPositionData(self.logic.lastValue)
+
+  def addWidgetMarkupObservers(self):
+      inputPath = self.ui.inputSelector.currentNode()
+      if inputPath is not None and inputPath.GetClassName() == "vtkMRMLMarkupsCurveNode":
+        self.widgetMarkupPointObserver = inputPath.AddObserver(slicer.vtkMRMLMarkupsNode.PointEndInteractionEvent, self.onWidgetMarkupPointEndInteraction)
+        self.widgetMarkupPointAddedObserver = inputPath.AddObserver(slicer.vtkMRMLMarkupsNode.PointAddedEvent, self.onWidgetMarkupPointAdded)
+        self.widgetMarkupPointRemovedObserver = inputPath.AddObserver(slicer.vtkMRMLMarkupsNode.PointRemovedEvent, self.onWidgetMarkupPointRemoved)
+        
+  def removeWidgetMarkupObservers(self, inputPath):
+    if inputPath is not None:
+        inputPath.RemoveObserver(self.widgetMarkupPointAddedObserver)
+        inputPath.RemoveObserver(self.widgetMarkupPointRemovedObserver)
+        inputPath.RemoveObserver(self.widgetMarkupPointObserver)
+        
+  def showCurrentPositionData(self, value):
+    # Get coordinates on path
+    currentPoint = self.logic.currentPosition(value);
+    if currentPoint.size == 0:
+        self.ui.locationLabel.setText("")
+        self.ui.distanceLabel.setText("")
+        self.ui.lengthLabel.setText("")
+        self.ui.orientationLabel.setText("")
+        return
+    position = "R " + str(int(currentPoint[0])) + ", " + "A " + str(int(currentPoint[1])) + ", " + "S " + str(int(currentPoint[2]))
+    self.ui.locationLabel.setText(position)
+    # Get path length
+    sizeOfArray = int(self.logic.cumDistancesArray.size)
+    length = str(round(self.logic.cumDistancesArray[sizeOfArray - 1], 1))
+    self.ui.lengthLabel.setText(length + " mm")
+    # Distance from start
+    distance = str(round(self.logic.cumDistancesArray[int(value)], 1))
+    self.ui.distanceLabel.setText(distance + " mm")
+    # VMTK centerline radius
+    inputPath = self.ui.inputSelector.currentNode()
+    if inputPath is not None and inputPath.GetClassName() == "vtkMRMLModelNode" and self.logic.vmtkCenterlineRadii.size > 0:
+        diameter = str(round(self.logic.vmtkCenterlineRadii[int(value)] * 2, 1))
+        self.ui.diameterLabel.setText(diameter + " mm")
+    # Orientation
+    orient = self.logic.getSliceOrientation()
+    orientation = "x " + str(round(orient[0], 1)) + "°,"
```

Comment:
VTK does not know what coordinate system the application is using. In Slicer, you know that the rendering coordinate systems are R, A, S axes. But for vascular applications specifically, you want to use RAO/LAO, CRA/CAU angles anyway.

Reply:
> In Slicer, you know that the rendering coordinate systems are R, A, S axes

OK, I'll change it to RAS instead of XYZ. It corresponds to the slice angles in Slicer's coordinate system. That of a C-Arm needs to be calculated and shown in a different field. I don't have enough insight to do that unfortunately.